### DISCUSSION TITLE
Cross-compiled strongswan can run normally, but there is a segment error when loading the configuration file？

### DISCUSSION THREAD

**Discussion by XYQ202** [strongswan/strongswan#1547] [Q&A]
Hi, Strongswan Teams,
          I have a problem, as follows:
          
![企业微信截图_16769742897077](https://user-images.githubusercontent.com/97773232/220323204-fb9d8569-e178-426f-8502-0df50c3a59f1.png)
         When I finished running ipsec start, I started strongswan, but when I loaded the configuration file, a segment error occurred, which does not seem to affect the normal loading of the configuration file. What is the reason and can it be avoided?

**Comment by tobiasbrunner** [comment]
You might want to attach a debugger to get a backtrace. Definitely make sure you don't mix plugins, libraries and executables from different versions/builds. Also, you are mixing `swanctl` and the legacy `ipsec` script (not really a problem or related to the problem, but it could indicate that you don't know what you are doing).

**Reply by XYQ202** [reply]
> You might want to attach a debugger to get a backtrace. Definitely make sure you don't mix plugins, libraries and executables from different versions/builds. Also, you are mixing `swanctl` and the legacy `ipsec` script (not really a problem or related to the problem, but it could indicate that you don't know what you are doing).

Will this affect the construction of tunnels?

**Reply by tobiasbrunner** [reply]
What do you mean?

**Reply by XYQ202** [reply]
> What do you mean?

I re-compiled and solved the problem.

**Comment by XYQ202** [comment]
I would also like to ask, at present, I have added the log function in strongswan.conf, but when the tunnel has been established, I will always write logs. Do you have the function of regularly clearing logs?
![企业微信截图_16771242437263](https://user-images.githubusercontent.com/97773232/220817544-42550f22-2c2e-4aae-8f33-3003ac2bd84a.png)

**Reply by tobiasbrunner** [reply]
Please [read the docs](https://docs.strongswan.org/docs/5.9/config/logging.html#_configuration_in_strongswan_conf) (in particular the part about SIGHUP).

**Reply by XYQ202** [reply]
> Please [read the docs](https://docs.strongswan.org/docs/5.9/config/logging.html#_configuration_in_strongswan_conf) (in particular the part about SIGHUP).

You mean to refresh the log through the command **swanctl --reload settings**. My idea is to delete the log at a fixed time, because I found that the strongswan log has occupied 200 MB of memory, which is very large. I want to clear the log regularly

**Reply by tobiasbrunner** [reply]
Yes, you could use `logrotate` to automate that, or do it manually.

**Reply by XYQ202** [reply]
> Yes, you could use `logrotate` to automate that, or do it manually.

Where is the relevant configuration of _**logrotate**_? Can you show me? Thank you!

**Reply by XYQ202** [reply]
> logrotate

Logrotate is a tool on Linux system. It cannot be configured in strongswan to manage logs, right?

**Reply by Thermi** [reply]
logrotate is usually run periodically through either (ana)cron or systemd timers. You just need to put the right configuration into `/etc/logrotate.d/` (assuming you have it installed). The distribution you are using probably already preinstalls it, or when you install it, you get that directory and some configuration files in it.

And yes, you can not configure strongSwan to manage the logs. It has no such functionality.
